### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/zig/commit/27ebd9fd73e470993222cf579497871e19f3c04e#commitcomment-56999974

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/zig/commit/c8c94bf5649d548d0808a464e7ef5261e08d6c1c

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

### CODE REVIEW

File: ci/azure/windows_msvc_script.bat [hunk 1]
```diff
@@ -1,7 +1,7 @@
 @echo on
 SET "SRCROOT=%cd%"
 SET "PREVPATH=%PATH%"
-SET "PREVMSYSEM=%MSYSTEM%"
+SET "PREVMSYSTEM=%MSYSTEM%"
```
Comment (jsoref) [inline ci/azure/windows_msvc_script.bat:4]: This appears to be a bug fix (see line 10 which expects this variable)

File: doc/langref.html.in [hunk 2]
```diff
@@ -7669,7 +7669,7 @@ test "main" {
 }
       {#code_end#}
       <p>
-      will ouput:
+      will output:
```
Comment (jsoref) [inline doc/langref.html.in:7672]: user doc facing?

File: src/codegen/spirv/spec.zig [hunk 3]
```diff
@@ -1302,7 +1302,7 @@ pub const Decoration = enum(u32) {
     RegisterINTEL = 5825,
     MemoryINTEL = 5826,
     NumbanksINTEL = 5827,
-    BankwidthINTEL = 5828,
```
Comment (jsoref) [inline src/codegen/spirv/spec.zig:1305]: ?

File: src/codegen/spirv/spec.zig [hunk 4]
```diff
@@ -1302,7 +1302,7 @@ pub const Decoration = enum(u32) {
...
+    BandwidthINTEL = 5828,
```
Comment (jsoref) [inline src/codegen/spirv/spec.zig:1305]: This feels like an API break

File: src/stage1/ir_print.cpp [hunk 5]
```diff
@@ -558,7 +558,7 @@ const char* ir_inst_gen_type_str(Stage1AirInstId id) {
         case Stage1AirInstIdWasmMemoryGrow:
             return "GenWasmMemoryGrow";
         case Stage1AirInstIdExtern:
-            return "GenExtrern";
+            return "GenExtern";
```
Comment (jsoref) [inline src/stage1/ir_print.cpp:561]: internal?

File: src/stage1/ir_print.cpp [hunk 6]
```diff
@@ -829,7 +829,7 @@ static const char *cast_op_str(CastOp op) {
         case CastOpIntToFloat: return "IntToFloat";
         case CastOpFloatToInt: return "FloatToInt";
         case CastOpBoolToInt: return "BoolToInt";
-        case CastOpNumLitToConcrete: return "NumLitToConcrate";
+        case CastOpNumLitToConcrete: return "NumLitToConcrete";
```
Comment (jsoref) [inline src/stage1/ir_print.cpp:832]: ditto?

File: src/stage1/target.cpp [hunk 7]
```diff
@@ -40,7 +40,7 @@ static const ZigLLVM_ArchType arch_list[] = {
     ZigLLVM_riscv64,        // RISC-V (64-bit): riscv64
     ZigLLVM_sparc,          // Sparc: sparc
     ZigLLVM_sparcv9,        // Sparcv9: Sparcv9
-    ZigLLVM_sparcel,        // Sparc: (endianness = little). NB: 'Sparcle' is a CPU variant
+    ZigLLVM_sparcel,        // Sparc: (endianness = little). NB: 'Sparcel' is a CPU variant
```
Comment (jsoref) [inline src/stage1/target.cpp:43]: I should have checked, this is wrong.
https://en.wikipedia.org/wiki/Sparcle

File: src/zig_clang.h [hunk 8]
```diff
@@ -934,10 +934,10 @@ enum ZigClangPreprocessedEntity_EntityKind {
...
 
 enum ZigClangExpr_ConstantExprKind {
-    ZigClangExpr_ContantExprKind_Normal,
-    ZigClangExpr_ContantExprKind_NonClassTemplateArgument,
-    ZigClangExpr_ContantExprKind_ClassTemplateArgument,
-    ZigClangExpr_ContantExprKind_ImmediateInvocation,
+    ZigClangExpr_ConstantExprKind_Normal,
+    ZigClangExpr_ConstantExprKind_NonClassTemplateArgument,
+    ZigClangExpr_ConstantExprKind_ClassTemplateArgument,
+    ZigClangExpr_ConstantExprKind_ImmediateInvocation,
```
Comment (jsoref) [inline src/zig_clang.h:940]: api break?

File: src/zig_clang_driver.cpp [hunk 9]
```diff
@@ -558,7 +558,7 @@ int ZigClang_main(int argc_, const char **argv_) {
   if (!UseNewCC1Process && IsCrash) {
     // When crashing in -fintegrated-cc1 mode, bury the timer pointers, because
     // the internal linked list might point to already released stack frames.
-    llvm::BuryPointer(llvm::TimerGroup::aquireDefaultGroup());
+    llvm::BuryPointer(llvm::TimerGroup::acquireDefaultGroup());
```
Comment (jsoref) [inline src/zig_clang_driver.cpp:561]: I need to drop this. llvm has been on my todo list for a long time, but I haven't gotten around to it.

File: src/type.zig [hunk 10]
```diff
@@ -3421,7 +3421,7 @@ pub const Type = extern union {
         anyerror_void_error_union,
         generic_poison,
         /// This is a special type for variadic parameters of a function call.
-        /// Casts to it will validate that the type can be passed to a c calling convetion function.
+        /// Casts to it will validate that the type can be passed to a c calling convection function.
```
Comment (Aransentin) [inline src/type.zig:3424]: "Convention", not "convection".

Comment (jsoref) [reply src/type.zig]: Oops, when Google Sheets made that suggestion, I was pretty sure it was wrong, I should have looked. Thanks, fixed.

File: src/codegen/spirv/spec.zig (hunk 3 above)
Comment (Vexu) [reply src/codegen/spirv/spec.zig:1305]: `BankwidthINTEL` is correct https://www.khronos.org/registry/SPIR-V/specs/unified1/SPIRV.html.

File: src/stage1/ir_print.cpp (hunk 5 above)
Comment (Vexu) [reply src/stage1/ir_print.cpp:561]: These are just debug output, fixing spelling here is fine.

File: src/zig_clang.h (hunk 8 above)
Comment (Vexu) [reply src/zig_clang.h:940]: Nope, as long as the numbers stay the same it's fine.

File: src/codegen/spirv/spec.zig (hunk 4 above)
Comment (Snektron) [reply src/codegen/spirv/spec.zig]: Probably intel jargon

Comment (jsoref) [reply src/codegen/spirv/spec.zig]: Yeah, it's bank, it's unfortunate that it isn't `BankWidthINTEL` to match `BankBitsINTEL`:

Number | Decoration
-- | --
5827 | NumbanksINTEL
5828 | BankwidthINTEL
5835 | BankBitsINTEL

**Comment by andrewrk** [comment]
Thanks!

I'm not interested in the action, but a periodic (once per year?) PR to fix these things in bulk would be welcome.

**Comment by jsoref** [comment]
@andrewrk: hmm. I'm not actively using zig. I ran across it somewhere and decided to try the tooling on it. I'm more likely to do periodic updates for projects that I do use (and even those probably get refreshes 18-30 months).

But, I'm up for experimenting. I've left a PR as a note to myself (and so others can see it). -- Each time I use my software in new ways, I learn things about it, and get opportunities to improve it. -- So this was helpful and will continue to be so.

Also note that there's really nothing particularly special about either the action or the tools I use to convert its word list into corrections. So, other people are more than welcome to use it to aim for your schedule.

Anyway, if someone pings me in a year, it's generally relatively easy for me to update (and as long as I have a repo that's keeping the metadata, and the repository has a baseline, the costs tend to be significantly lower than starting from scratch).

**Pull request merged**